service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -191,23 +192,22 @@ protected Router createRouter() {
 
         // route (failure) handlers are added in a specific order here!
         final Router router = Router.router(vertx);
+        if (httpServiceConfigProperties.isAuthenticationRequired()) {
+            // AuthHandler
+            addAuthHandler(router);
+        }
         final Route matchAllFailuresRoute = router.route();
         // failure handler is added on a separate route for which no name is set, otherwise all tracing spans
         // of failed HTTP requests would get that route name as span name
         matchAllFailuresRoute.failureHandler(new DefaultFailureHandler());
 
         final Route matchAllRoute = router.route();
+
         // route name will be used as HTTP request tracing span name,
         // ensuring a fixed name is set in case no other route matches (otherwise the span name would unsuitably be set to the request path)
         matchAllRoute.setName("/* (default route)");
         // 1. handler to keep track of the tracing span created by the Vert.x/Quarkus instrumentation (set as active span there)
         matchAllRoute.handler(HttpServerSpanHelper.getRouteHandlerForAdoptingActiveSpan(tracer, getCustomTags()));
-        // 2. BodyHandler with request size limit
```

Comment:
The removal of the BodyHandler here means that the specific routes added later (via `endpoint.addRoutes()` or `addCustomRoutes()`) need to use a BodyHandler if needed. Looking at the places where `endpoint.addRoutes()` is implemented (e.g. in `DelegatingCredentialsManagementHttpEndpoint`), the `bodyHandler.setUploadsDirectory(DEFAULT_UPLOADS_DIRECTORY)` invocation isn't done there (`HttpServiceBase#DEFAULT_UPLOADS_DIRECTORY` isn't used any more).

FMPOV, the places where the BodyHandler gets used in the `endpoint.addRoutes()` implementations should be adapted to also set the uploads directory there.